src/include/btree_inline.h

```diff
@@ -2054,20 +2050,21 @@ __wt_btcur_skip_page(WT_CURSOR_BTREE *cbt)
      * We are making these decisions while holding a lock for the page as checkpoint or eviction can
      * make changes to the data structures (i.e., aggregate timestamps) we are reading.
      */
-    if (session->txn->isolation == WT_ISO_SNAPSHOT && !__wt_page_is_modified(page) &&
-      !F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE) && previous_state == WT_REF_MEM) {
-
+    if (session->txn->isolation == WT_ISO_SNAPSHOT && !ignore_tombstone &&
+      previous_state == WT_REF_MEM) {
         /* We only try to lock the page once. */
         if (!WT_REF_CAS_STATE(session, ref, previous_state, WT_REF_LOCKED))
-            return false;
+            return (0);
```

Comment:
Why only attempt to lock the page once, why not wait for the lock?

Reply:
For the benchmarks we have been looking at, we figured out that the cursor next and previous calls are the parts of the code where the application threads are spending more time. If we wait for the lock in a loop (i.e. use `WT_REF_LOCK()`), maybe it will introduce some regression. However, we are not sure as we don't have data about probability of failing to acquire the lock. 

@sulabhM - Maybe if we can add some debug stat to capture how many times we fail to acquire the lock, we will be able to decide between try-once or blocking wait. For example, not worth trying `WT_REF_LOCK()` if we are succeed the existing mechanism is able to lock the page 90% of the time.

Reply:
I will keep it as such for now. I will run a local test with block till locking, and see if it makes any difference. It will be hard to measure any perf difference we get comparing blocking vs try locking. I imagine a page is likely going through eviction/checkpoint we were unable to lock it, which means even if we decide to fall back to reading it, the read will have to wait. So, I suspect it wont make much difference in such a case.

Reply:
Note: I did a few local runs on the workload I am investigating. I did not see any significant difference between blocking for a lock or giving up try locking.